src/aeso_fcode_to_fate.erl

```diff
@@ -751,6 +734,66 @@ push(A) -> {'STORE', ?a, A}.
 tuple(0) -> push(?i({tuple, {}}));
 tuple(N) -> aeb_fate_ops:tuple(?a, N).
 
+dbg_loc(Env, Ann) ->
+    case proplists:get_value(debug_info, Env#env.options, false) of
+        false -> [];
+        true  ->
+            File = case proplists:get_value(file, Ann) of
+                       no_file -> "";
+                       F       -> F
+                   end,
+            Line = proplists:get_value(line, Ann),
+            case Line of
+                undefined -> [];
+                _         -> [{'DBG_LOC', {immediate, File}, {immediate, Line}}]
+            end
+    end.
+
+dbg_scoped_vars(_Env, [], SCode) ->
+    SCode;
+dbg_scoped_vars(Env, [Var | Rest], SCode) ->
+    dbg_scoped_vars(Env, Rest, dbg_scoped_var(Env, Var, SCode)).
+
+dbg_scoped_var(Env = #env{saved_fresh_names = SavedFreshNames}, Var, SCode) ->
+    case proplists:get_value(debug_info, Env#env.options, false) of
+        false -> SCode;
+        true  ->
+            case maps:get(Var, SavedFreshNames, Var) of
+                "%" ++ _ -> SCode;
```

Comment:
>what is %?

This is for fresh names (they have the form `%n` like `%0` or `%1`). We don't want to do have `DBG_DEF` and `DBG_UNDEF` for this fresh names.

>I would use some {atom, _} instead?

I don't get your suggestion. Can you explain this more?

Reply:
Ah I see. Maybe then have a function to tell if a variable is fresh? This piece of code looks super random if you don't know it. 

What I meant is to distinguish things like `{user_var, _}` and `{system_var, _}` (naming whatever), so it would be clear that we are handling a variable added by the system. But I think having a function to check it based on name would be simpler for now.

Reply:
Done here ad382302f84c895f71d2ee9ad3e4d2e59b5b4f0d.